db/system_keyspace.cc

```diff
@@ -1179,6 +1185,22 @@ schema_ptr system_keyspace::view_build_status_v2() {
     return schema;
 }
 
+schema_ptr system_keyspace::view_building_coordinator_tasks() {
+    static thread_local auto schema = [] {
+        auto id = generate_legacy_id(NAME, VIEW_BUILDING_COORDINATOR_TASKS);
+        return schema_builder(NAME, VIEW_BUILDING_COORDINATOR_TASKS, id)
+                .with_column("keyspace_name", utf8_type, column_kind::partition_key)
+                .with_column("view_name", utf8_type, column_kind::partition_key)
+                .with_column("host_id", uuid_type, column_kind::clustering_key)
+                .with_column("shard", int32_type, column_kind::clustering_key)
```

Comment:
Why do we need to store the host id and shard, aren't they volatile?

Reply:
We store task for shards on hosts, is there any other way to identify tasks' targets (host+shard) without those fields?

Reply:
Right now, reducing the shard count is not supported for tablet keyspaces: https://github.com/scylladb/scylladb/issues/16739. I'm not sure what the solution will be but it will most likely require redistributing sstables across shards and modifying information about shards in `system.tablets`. I suppose we should do something similar for the view builder tasks when we get to implement resharding.